executor/runtime/docker/docker.go

```diff
@@ -1427,6 +1436,98 @@ func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.
 	}
 }
 
+// startOtherUserContainers launches the other user containers, only looking
+// any container objects in the pod *after* the first one, converting the
+// v1.Container spec into something docker can understand, and then
+// running that container.
+func (r *DockerRuntime) startOtherUserContainers(ctx context.Context, pod *v1.Pod, mainContainerID string) error {
+	if pod == nil || pod.Spec.Containers == nil {
+		return nil
+	}
+	otherUserContainers := pod.Spec.Containers[1:]
+	if len(otherUserContainers) == 0 {
+		return nil
+	}
+	l := log.WithField("taskID", r.c.TaskID())
+	l.Infof("Starting up %d other user containers", len(otherUserContainers))
+	// Currently the order here is hard-coded to be whatever order it is in the podspec
+	// with no health or startup probes
+	for i, v1Container := range otherUserContainers {
+		l.Debugf("Starting other user container %d/%d: %s", i+1, len(otherUserContainers), v1Container.Name)
+		err := pullWithRetries(ctx, r.metrics, r.client, v1Container.Image, doDockerPull)
+		if err != nil {
+			return fmt.Errorf("Failed to pull an image for user container %s: %s", v1Container.Name, err)
+		}
+		err = r.startOtherUserContainer(ctx, v1Container, mainContainerID)
+		if err != nil {
+			return fmt.Errorf("Failed to create/start user container %s: %s", v1Container.Name, err)
+		}
+	}
+	return nil
+}
+
+func (r *DockerRuntime) startOtherUserContainer(ctx context.Context, v1Container v1.Container, mainContainerID string) error {
+	l := log.WithField("taskID", r.c.TaskID())
+	containerName := r.c.TaskID() + v1Container.Name
+	dockerContainerConfig, dockerHostConfig, dockerNetworkConfig := k8sContainerToDockerConfigs(v1Container, mainContainerID)
+	l.WithFields(map[string]interface{}{
+		"dockerCfg": logger.ShouldJSON(ctx, *dockerContainerConfig),
+		"hostCfg":   logger.ShouldJSON(ctx, *dockerHostConfig),
+	}).Info("Creating other user container in docker")
+	containerCreateBody, err := r.client.ContainerCreate(ctx, dockerContainerConfig, dockerHostConfig, dockerNetworkConfig, containerName)
```

Comment:
Same comment as above. We can pull and create these containers in parallel (alongside/ like sidecars) then launch them here.

Reply:
Yes